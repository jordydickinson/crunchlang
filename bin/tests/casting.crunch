fun main!() {
    var xs = {1, 2, 3};
    var xs_ptr: uint8* = xs as uint8*;
}
