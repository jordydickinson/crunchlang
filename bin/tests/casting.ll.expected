; ModuleID = 'casting.crunch'
source_filename = "casting.crunch"

@llvm.global_ctors = appending global [1 x void ()*] [void ()* @init.ctors]

define void @main() {
entry:
  br label %body

body:                                             ; preds = %entry
  %xs = alloca [3 x i32]
  store [3 x i32] [i32 1, i32 2, i32 3], [3 x i32]* %xs
  %xs_ptr = alloca i8*
  %xs.0 = getelementptr [3 x i32], [3 x i32]* %xs, i32 0, i32 0
  %xs.0.cast = bitcast i32* %xs.0 to i8*
  store i8* %xs.0.cast, i8** %xs_ptr
  ret void
}

define void @init.ctors() {
entry:
  ret void
}
